src/arrayMethodJoin.js

```diff
@@ -3,9 +3,40 @@
 /**
  * Implement method join
  */
+
+// - There is kinda an easy solution for
+// |-- this task but with some weird tests
+// |-- like null could be a separator
+// |-- but not an undefined, or NaN. XD
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let res = '';
+    let symbol = separator;
+
+    if (separator === null) {
+      symbol = 'null';
+    } else if (separator === '') {
```

Comment:
you can also set some value by default to the `separator` argument and avoid one of if / else statement

Reply:
done)
```
   funcName(separator = ',') {}
```